### PR TITLE
Create RoomService.java

### PR DESCRIPTION
**Pull request by NaltakyanArman**

### CODE REVIEW

File: src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    public List<Room> findAll(final Pageable pageable) {
+        return roomRepository.findAll(pageable).getContent();
+    }
+
+    public Room save(Room room) {
+        notNull(room, "Employee should not be null");
+        notNull(room.getType(), "room Type should not be null");
+        notNull(room.getBedNumbers(), "room BedNumber should not be null");
```
Comment (nareknaltakyansfl) [inline src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java:40]: please uppercase

File: src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        notNull(room.getPricePerDay(), "room PricePerDay should not be null");
+        if (room.getId() != null && existsById(room.getId())) {
+            throw new EntityExistsException("Room with id: " + room.getId() + " already exists");
+        }
+        return roomRepository.save(room);
+    }
+
+    public void update(Room room) {
+        notNull(room, "Employee should not be null");
```
Comment (nareknaltakyansfl) [inline src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java:49]: who is Eployee?

File: src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    private boolean existsById(Long id) {
+        return roomRepository.existsById(id);
+    }
+
+    public Room findById(Long id) {
+        Room room = roomRepository.findById(id).orElse(null);
+        if (room == null) {
+            throw new EntityNotFoundException("Cannot find Room  with id: " + id);
+        } else return room;
```
Comment (nareknaltakyansfl) [inline src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java:30]: why else?

File: src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            throw new EntityNotFoundException("Room find with id: " + room.getId());
+        }
+        roomRepository.save(room);
+    }
+
+    public void deleteById(Long id) {
+        if (!existsById(id)) {
+            throw new EntityNotFoundException("Room find with id: " + id);
+        }
+        roomRepository.deleteById(id);
```
Comment (nareknaltakyansfl) [inline src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java:63]: do not hard delete row, only update by setting deleted's value

File: src/main/java/com/naltakyan/hotelmanagement/repository/RoomRepository.java [hunk 5]
```diff
@@ -1,8 +1,18 @@
...
 import org.springframework.stereotype.Repository;
 
+import java.util.Optional;
+
 @Repository
-public interface RoomRepository extends JpaRepository<Room, Long> {}
+public interface RoomRepository extends JpaRepository<Room, Long> {
+    Optional<Room> findByIdAndDeletedIsNull(Long id);
+
+    Page<Room> getRoomServicesByDeletedIsNull(Pageable pageable);
```
Comment (nareknaltakyan1) [inline src/main/java/com/naltakyan/hotelmanagement/repository/RoomRepository.java:17]: getRoomsByDeletedIsNull

File: src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+public class RoomService {
+
+    private final RoomRepository roomRepository;
+
+    private boolean existsById(Long id) {
+        return roomRepository.existsById(id);
+    }
+
+    public Room findById(Long id) {
+        Room room = roomRepository.findById(id).orElse(null);
```
[the remark this answers is not in the document]
Comment (nareknaltakyan1) [reply src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java:28]: use findByIdAndDeletedIsNull

File: src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Transactional
+    public void update(Room room) {
+        notNull(room, "Room should not be null");
+        notNull(room.getType(), "Room Type should not be null");
+        notNull(room.getBedNumbers(), "Room BedNumber should not be null");
+        notNull(room.getPricePerDay(), "Room PricePerDay should not be null");
+        if (!existsById(room.getId())) {
+            throw new EntityNotFoundException("Room find with id: " + room.getId());
```
Comment (nareknaltakyan1) [inline src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java:59]: not Room find but Cannot find

File: src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Transactional
+    public Room save(Room room) {
+        notNull(room, "Room should not be null");
+        notNull(room.getType(), "Room Type should not be null");
+        notNull(room.getBedNumbers(), "Room BedNumber should not be null");
+        notNull(room.getPricePerDay(), "Room PricePerDay should not be null");
+        if (room.getId() != null && existsById(room.getId())) {
+            throw new EntityExistsException("Room with id: " + room.getId() + " already exists");
```
Comment (nareknaltakyan1) [inline src/main/java/com/naltakyan/hotelmanagement/service/RoomService.java:47]: not Room find but Cannot find Room

**Pull request merged**